client/resource_group/controller/model.go

```diff
@@ -205,3 +212,26 @@ func sub(custom1 *rmpb.Consumption, custom2 *rmpb.Consumption) {
 	custom1.KvReadRpcCount -= custom2.KvReadRpcCount
 	custom1.KvWriteRpcCount -= custom2.KvWriteRpcCount
 }
+
+// getSQLProcessCPUTime returns the cumulative user+system time (in ms) since the process start.
+func getSQLProcessCPUTime(isServerless bool) float64 {
+	if isServerless {
+		return getSysProcessCPUTime()
+	}
+	return getGroupProcessCPUTime()
+}
+
+func getSysProcessCPUTime() float64 {
+	pid := os.Getpid()
+	cpuTime := gosigar.ProcTime{}
+	if err := cpuTime.Get(pid); err != nil {
+		log.Error("getCPUTime get pid failed", zap.Error(err))
+	}
+
+	return float64(cpuTime.User + cpuTime.Sys)
+}
+
+// TODO: will be implemented by Goroutine after https://github.com/golang/go/issues/41554 merged
```

Comment:
no need to rely on it, we can do it in another way first.

Reply:
Can you be more detailed?